### PR TITLE
citron: init at 0.15.0

### PR DESCRIPTION
**Pull request by VuiMuich**
## Description of changes

`citron` is a tool to send sytemdata (like battery, backlight or memory) via dbust to a notification daemon.
https://git.sr.ht/~grtcdr/citron

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -376,6 +376,8 @@ with pkgs;
 
   circumflex = callPackage ../applications/networking/circumflex { };
 
+  citorn = callPackage ../tools/misc/citron { };
```
Comment (h7x4) [inline pkgs/top-level/all-packages.nix:379]: :)

```suggestion
  citron = callPackage ../tools/misc/citron { };
```

Comment (VuiMuich) [reply pkgs/top-level/all-packages.nix:379]: Oh my gosh this is an embarrassing one 😳

**Review by h7x4 — APPROVED** [review]
Built and ran binary. Seems to work fine. Nice!

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (VuiMuich) [reply pkgs/top-level/all-packages.nix:379]: btw. fix for this typo is merged and squashed.

**Comment by VuiMuich** [comment]
> Built and ran binary. Seems to work fine. Nice!

Thanks for the reiew.

As per comment in #251335, I changed `sha256` to `hash` and `cargoSha256` to `cargoHash` as this seems to be preferred.

**Review by cafkafk — CHANGES_REQUESTED** [review]
I asked in the nixpkgs matrix room but they were slow too reply so I'll just drop it here and then if I'm wrong I'm sorry, but I'm pretty sure it would be best if you changed the order of your commits, so you first add yourself to the maintainers file, and then add the package.

**Comment by VuiMuich** [comment]
> I asked in the nixpkgs matrix room but they were slow too reply so I'll just drop it here and then if I'm wrong I'm sorry, but I'm pretty sure it would be best if you changed the order of your commits, so you first add yourself to the maintainers file, and then add the package.

Thanks to lazygit this was super simple :+1: 

Thanks for mentioning.

**Comment by cafkafk** [comment]
Result of `nixpkgs-review pr 250939` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>citron</li>
  </ul>
</details>

**Review by cafkafk — APPROVED** [review]
Tested locally on x86_64 GNU/Linux, binary built as expected, and shows help. Didn't test functionality of binary.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/citron/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{
+  dbus
+, fetchCrate
+, installShellFiles
+, lib
+, pkg-config
+, rustPlatform
+}:
+
```
[the remark this answers is not in the document]
Comment (VuiMuich) [reply pkgs/tools/misc/citron/default.nix:9]: I don't mind to change that, but what is the actual preference here? In my first PR #247425 the reviewer reordered explicitly alphabeticaly.

[the remark this answers is not in the document]
Comment (h7x4) [reply pkgs/tools/misc/citron/default.nix]: I guess there's the last point at https://nixos.org/manual/nixpkgs/stable/#sec-syntax

[the remark this answers is not in the document]
Comment (VuiMuich) [reply pkgs/tools/misc/citron/default.nix]: Thanks!

[the remark this answers is not in the document]
Comment (VuiMuich) [reply pkgs/tools/misc/citron/default.nix]: Should this be added to https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#code-conventions or am I just overlooking it?

File: pkgs/tools/misc/citron/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  nativeBuildInputs = [ installShellFiles pkg-config ];
+
+  postInstall = ''
+    installManPage doc/citron.1
+  '';
+
+  meta = with lib; {
+    homepage = "https://git.sr.ht/~grtcdr/citron";
+    description = "system data via on-demand notifications";
```
Comment (AndersonTorres) [inline pkgs/tools/misc/citron/default.nix:30]:
```suggestion
    description = "System data via on-demand notifications";
```

File: pkgs/tools/misc/citron/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+  buildInputs = [ dbus ];
+
+  nativeBuildInputs = [ installShellFiles pkg-config ];
+
+  postInstall = ''
+    installManPage doc/citron.1
+  '';
+
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/tools/misc/citron/default.nix:28]:
```suggestion
  meta = {
```

File: pkgs/tools/misc/citron/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    homepage = "https://git.sr.ht/~grtcdr/citron";
+    description = "system data via on-demand notifications";
+    license = licenses.mit;
+    maintainers = with maintainers; [ vuimuich ];
```
Comment (AndersonTorres) [inline pkgs/tools/misc/citron/default.nix:32]:
```suggestion
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ vuimuich ];
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/citron/default.nix (hunk 3 above)
Comment (VuiMuich) [reply pkgs/tools/misc/citron/default.nix]: Docs are a bit ambiguous on the first letter capitalisation in the description string:
https://nixos.org/manual/nixpkgs/stable/#var-meta-description

> description
> A short (one-line) description of the package. This is shown by nix-env -q --description and also on the Nixpkgs release pages.
>
>Don’t include a period at the end. Don’t include newline characters. Capitalise the first character. For brevity, don’t repeat > the name of package — just describe what it does.
>
> Wrong: "libpng is a library that allows you to decode PNG images."
> 
> Right: "A library for decoding PNG images"

File: pkgs/tools/misc/citron/default.nix (hunk 4 above)
Comment (h7x4) [reply pkgs/tools/misc/citron/default.nix]: Could I ask about the motivation behind these changes?

**Review by AndersonTorres — APPROVED** [review]

**Comment by AndersonTorres** [comment]
Ping me 2 days to merge

**Comment by VuiMuich** [comment]
@AndersonTorres here goes your ping.
Also I just rebased, idk how relevant or necessary it would be, but thought it wouldn't hurt either, right?

**Comment by AndersonTorres** [comment]
Way better.

**Comment by VuiMuich** [comment]
Thanks for the merge!

**Comment by devhell** [comment]
esult of `nixpkgs-review pr 250939` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>citron</li>
  </ul>
</details>

Tested binary, and works as expected (except `citron network`, but that might be due to my setup).

**Comment by VuiMuich** [comment]
> esult of `nixpkgs-review pr 250939` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 
> <details>
> 
>   <summary>1 package built:</summary>
> 
>   <ul>
> 
>     <li>citron</li>
> 
>   </ul>
> 
> </details>
> 
> 
> 
> Tested binary, and works as expected (except `citron network`, but that might be due to my setup).

Thanks for testing.
I typically don't use the networking feature, but iirc, it requires the ability interface being defined in a config file placed in XDG_CONFIG_DIR.
Maybe it's worth to propose some sane fallback upstream, if no interface is configured.

**Pull request merged**